### DISCUSSION TITLE
Pending defaultValue in atom/fetchAtom

### DISCUSSION THREAD

**Discussion by janrembold** [pmndrs/jotai#1394] [Q&A] [answered]
I'm searching for a best practice to avoid the initial first render when the atom is still `null` shortly before it starts fetching and being a pending Promise. That means I see a first render before the component jumps into Suspense mode. 

Sadly, in my case there is a parameter necessary to fetch the atoms value (not very good reflected in the demo below). So its not possible to simply set the fetch directly as default value, because the parameter is a property from outside the component.

I could simply set `if(!foo) return null` or do something else to avoid that first render, but what would be a best practice to not use such a manual tweak and instead fully use the power of Jotai and Reacts Suspense?

Codesandbox: https://codesandbox.io/s/react-typescript-forked-jbk2me?file=/src/App.tsx 
``` typescript
interface Foo {
  test: string;
}

const getFoo = (test: string): Promise<Foo> => {
  return new Promise((resolve) => {
    setTimeout(() => resolve({ test }), 2000);
  });
};

const fooAtom = atom<Foo | Promise<Foo> | null>(null);
const fetchFooAtom = atom(
  (get) => get(fooAtom),
  async (_get, set, param: string) => set(fooAtom, getFoo(param))
);

const Test: React.FC = () => {
  const [foo, fetchFoo] = useAtom(fetchFooAtom);

  useEffect(() => {
    fetchFoo("bar");
  }, []);

  console.log("render: foo", foo?.test || "not loaded yet");

  return <div>foo: {foo?.test || "null"}</div>;
};

export default function App() {
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <Suspense fallback="Suspense kicks in...">
        <Test />
      </Suspense>
    </div>
  );
}
```

console logs:
```
# render: foo not loaded yet
// now Suspense kicks in
# render: foo bar
```

Thanks in advance and @dai-shi you're doing a fantastic job here!!!!!
Best, Jan

**Comment by dai-shi** [comment, marked as answer]
If I understand correctly, you want to have an atom that suspends immediately, and forever until you explicitly set it.
Some pointers:
- https://jotai.org/docs/guides/async#async-forever
- https://github.com/jotai-labs/jotai-suspense 's `atomWithPending`
  - https://github.com/jotai-labs/jotai-suspense/pull/1

**Reply by janrembold** [reply]
Thanks! I just didn't see the `async-forever` docs 👍 

But this also means I need to trigger the `fetchFoo('bar')` method from outside the component as it would never reach the `useEffect` that currently triggers it. Anyways, this leads in the absolute right direction!!! Thank you very much!
